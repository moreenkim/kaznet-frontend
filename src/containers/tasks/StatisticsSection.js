import React, { Component } from "react";
import FontAwesomeIcon from "@fortawesome/react-fontawesome";
import { Col, Row } from "reactstrap";

export default class StatisticsSection extends Component {
  render() {
    return (
      <Col sm="12" md={{ size: 8, offset: 2 }}>
        <Row>
          <Col sm="3" className="kaznet-sub">
            <p className="kaznet-stats-count">
              {this.props.totalSubmissions} Submissions
            </p>
          </Col>
          <Col sm="9" className="kaznet-sub">
            {this.props.task && this.props.task.attributes.xform_title ? (
              <a
                href={this.props.formURL}
                className="btn btn-primary text-white"
                target="_blank"
              >
                <FontAwesomeIcon icon="clone" /> Review all submissions
              </a>
            ) : (
              ""
            )}
          </Col>
        </Row>
        <Row>
          <Col sm="6">
            <Row>
              <Col sm="6" className="kaznet-no-gutter">
                <div className="kaznet-stats">
                  <p className="stats-data">{this.props.accepted}</p>
                  <p className="stats-header">
                    Accepted
                    <FontAwesomeIcon
                      icon="external-link-alt"
                      className="fa-xs icon-link withspace"
                    />
                  </p>
                </div>
              </Col>
              <Col sm="6" className="kaznet-no-gutter left-fix">
                <div className="kaznet-stats">
                  <p className="stats-data">{this.props.reward}</p>
                  <p className="stats-header">
                    Total Reward
                    <FontAwesomeIcon
                      icon="laptop"
                      className="fa-xs icon-link withspace"
                    />
                  </p>
                </div>
              </Col>
            </Row>
          </Col>
          <Col sm="6" className="overflow-fix">
            <Row>
              <Col sm="6" className="kaznet-stats-col">
                <div className="kaznet-stats">
                  <h1 className="stats-data">{this.props.review}</h1>
                  <p className="stats-header">
                    Needs Review
                    <FontAwesomeIcon
                      icon="external-link-alt"
                      className="fa-xs icon-link withspace"
                    />
                  </p>
                </div>
              </Col>
              <Col sm="6" className="kaznet-stats-col">
                <div className="kaznet-stats overflow-fix">
                  <h1 className="stats-data">{this.props.rejected}</h1>
                  <p className="stats-header">
                    Rejected
                    <FontAwesomeIcon
                      icon="external-link-alt"
                      className="fa-xs icon-link withspace"
                    />
                  </p>
                </div>
              </Col>
            </Row>
          </Col>
        </Row>
        <hr />
      </Col>
    );
  }
}
