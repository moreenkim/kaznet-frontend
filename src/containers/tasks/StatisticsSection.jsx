import React from 'react';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import { Col, Row } from 'reactstrap';
import PropTypes from 'prop-types';
import ExportModal from '../../components/ExportModal';

const StatisticsSection = props => {
  const {
    downloadModalHandler,
    task,
    formURL,
    xformTableURL,
    accepted,
    totalSubmissions,
    reward,
    review,
    rejected,
    modalState,
    className,
    onFormSubmit,
    taskName
  } = props;

  return (
    <Col sm="12" md={{ size: 8, offset: 2 }}>
      <Row>
        <Col sm="3" className="kaznet-sub">
          <p className="kaznet-stats-count">
            {totalSubmissions}
            &nbsp;Submissions
          </p>
        </Col>
        <Col sm="9" className="kaznet-sub">
          {task && task.attributes.xform_title ? (
            <a
              href={formURL}
              className="btn btn-primary text-white"
              target="_blank"
              rel="noopener noreferrer"
            >
              <FontAwesomeIcon icon="clone" />
              Review all submissions
            </a>
          ) : (
            ''
          )}
        </Col>
      </Row>
      <Row>
        <Col sm="6">
          <Row>
            <Col sm="6" className="kaznet-no-gutter">
              <div className="kaznet-stats">
                <p className="stats-data">{accepted}</p>
                <p className="stats-header">
                  Accepted
                  {xformTableURL !== null && (
                    <a href={xformTableURL} target="_blank" rel="noopener noreferrer">
                      <FontAwesomeIcon
                        icon="external-link-alt"
                        className="fa-xs icon-link withspace"
                      />
                    </a>
                  )}
                </p>
              </div>
            </Col>
            <Col sm="6" className="kaznet-no-gutter left-fix">
              <div className="kaznet-stats">
                <p className="stats-data">{reward}</p>
                <p className="stats-header">
                  Total Reward
                  {accepted > 0 && (
                    <FontAwesomeIcon
                      icon="laptop"
                      className="fa-xs icon-link withspace clickable-icon"
                      onClick={downloadModalHandler}
                    />
                  )}
                </p>
              </div>
            </Col>
          </Row>
        </Col>
        <Col sm="6" className="overflow-fix">
          <Row>
            <Col sm="6" className="kaznet-stats-col">
              <div className="kaznet-stats">
                <h1 className="stats-data">{review}</h1>
                <p className="stats-header">
                  Needs Review
                  {xformTableURL !== null && (
                    <a href={xformTableURL} target="_blank" rel="noopener noreferrer">
                      <FontAwesomeIcon
                        icon="external-link-alt"
                        className="fa-xs icon-link withspace"
                      />
                    </a>
                  )}
                </p>
              </div>
            </Col>
            <Col sm="6" className="kaznet-stats-col">
              <div className="kaznet-stats overflow-fix">
                <h1 className="stats-data">{rejected}</h1>
                <p className="stats-header">
                  Rejected
                  {xformTableURL !== null && (
                    <a href={xformTableURL} target="_blank" rel="noopener noreferrer">
                      <FontAwesomeIcon
                        icon="external-link-alt"
                        className="fa-xs icon-link withspace"
                      />
                    </a>
                  )}
                </p>
              </div>
            </Col>
          </Row>
          <ExportModal
            modalState={modalState}
            downloadModalHandler={downloadModalHandler}
            className={className}
            onFormSubmit={onFormSubmit}
            name={taskName}
          />
        </Col>
      </Row>
      <hr />
    </Col>
  );
};

StatisticsSection.propTypes = {
  downloadModalHandler: PropTypes.func.isRequired,
  task: PropTypes.shape({
    attributes: PropTypes.shape({
      xform_title: PropTypes.string
    })
  }).isRequired,
  formURL: PropTypes.string.isRequired,
  xformTableURL: PropTypes.string.isRequired,
  accepted: PropTypes.number.isRequired,
  totalSubmissions: PropTypes.number.isRequired,
  reward: PropTypes.number.isRequired,
  review: PropTypes.number.isRequired,
  rejected: PropTypes.number.isRequired,
  modalState: PropTypes.bool.isRequired,
  className: PropTypes.string.isRequired,
  onFormSubmit: PropTypes.func.isRequired,
  taskName: PropTypes.string.isRequired
};

export default StatisticsSection;
