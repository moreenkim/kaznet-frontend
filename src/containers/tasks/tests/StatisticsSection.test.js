// Test StatisticsSection
import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import StatisticsSection from "../StatisticsSection";
import * as fixtures from "../../../store/tasks/tests/fixtures";

describe("containers/task/StatisticsSection", () => {
  it("renders without crashing", () => {
    shallow(<StatisticsSection />);
  });

  it("renders statistics correctly", () => {
    const task = fixtures.singleTask;
    const wrapper = mount(
      <StatisticsSection
        accepted={task.attributes.approved_submissions_count}
        reward={task.attributes.total_bounty_payout}
        review={task.attributes.pending_submissions_count}
        rejected={task.attributes.rejected_submissions_count}
        totalSubmissions={task.attributes.submission_count}
        task={task}
        formURL="https://ona.io/kaznettest/64730/327942"
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
