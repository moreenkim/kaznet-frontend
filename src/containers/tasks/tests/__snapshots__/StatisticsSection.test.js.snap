// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/task/StatisticsSection renders statistics correctly 1`] = `
<StatisticsSection>
  <Col
    md={
      Object {
        "offset": 2,
        "size": 8,
      }
    }
    sm="12"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <div
      className="col-sm-12 col-md-8 offset-md-2"
    >
      <Row
        tag="div"
      >
        <div
          className="row"
        >
          <Col
            className="kaznet-sub"
            sm="3"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="kaznet-sub col-sm-3"
            >
              <p
                className="kaznet-stats-count"
              >
                 Submissions
              </p>
            </div>
          </Col>
          <Col
            className="kaznet-sub"
            sm="9"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="kaznet-sub col-sm-9"
            >
              <Button
                color="primary"
                tag="button"
              >
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon="clone"
                    listItem={false}
                    mask={null}
                    name=""
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-clone fa-w-16 "
                      data-icon="clone"
                      data-prefix="fas"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M464 0c26.51 0 48 21.49 48 48v288c0 26.51-21.49 48-48 48H176c-26.51 0-48-21.49-48-48V48c0-26.51 21.49-48 48-48h288M176 416c-44.112 0-80-35.888-80-80V128H48c-26.51 0-48 21.49-48 48v288c0 26.51 21.49 48 48 48h288c26.51 0 48-21.49 48-48v-48H176z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                   Review all submissions
                </button>
              </Button>
            </div>
          </Col>
        </div>
      </Row>
      <Row
        tag="div"
      >
        <div
          className="row"
        >
          <Col
            sm="6"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col-sm-6"
            >
              <Row
                tag="div"
              >
                <div
                  className="row"
                >
                  <Col
                    className="kaznet-no-gutter"
                    sm="6"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    <div
                      className="kaznet-no-gutter col-sm-6"
                    >
                      <div
                        className="kaznet-stats"
                      >
                        <p
                          className="stats-data"
                        />
                        <p
                          className="stats-header"
                        >
                          Accepted
                        </p>
                      </div>
                    </div>
                  </Col>
                  <Col
                    className="kaznet-no-gutter left-fix"
                    sm="6"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    <div
                      className="kaznet-no-gutter left-fix col-sm-6"
                    >
                      <div
                        className="kaznet-stats"
                      >
                        <p
                          className="stats-data"
                        />
                        <p
                          className="stats-header"
                        >
                          Total Reward
                        </p>
                      </div>
                    </div>
                  </Col>
                </div>
              </Row>
            </div>
          </Col>
          <Col
            className="overflow-fix"
            sm="6"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="overflow-fix col-sm-6"
            >
              <Row
                tag="div"
              >
                <div
                  className="row"
                >
                  <Col
                    className="kaznet-stats-col"
                    sm="6"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    <div
                      className="kaznet-stats-col col-sm-6"
                    >
                      <div
                        className="kaznet-stats"
                      >
                        <h1
                          className="stats-data"
                        />
                        <p
                          className="stats-header"
                        >
                          Needs Review
                        </p>
                      </div>
                    </div>
                  </Col>
                  <Col
                    className="kaznet-stats-col"
                    sm="6"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    <div
                      className="kaznet-stats-col col-sm-6"
                    >
                      <div
                        className="kaznet-stats overflow-fix"
                      >
                        <h1
                          className="stats-data"
                        />
                        <p
                          className="stats-header"
                        >
                          Rejected
                        </p>
                      </div>
                    </div>
                  </Col>
                </div>
              </Row>
            </div>
          </Col>
        </div>
      </Row>
      <hr />
    </div>
  </Col>
</StatisticsSection>
`;
